operations-stdlib/src/commonMain/kotlin/string/Replace.kt

```diff
@@ -0,0 +1,86 @@
+package string
+
+import operation.StandardLogicOperation
+import kotlin.runCatching
+import utils.asList
+
+object Replace: StandardLogicOperation, StringUnwrapStrategy {
+    override fun evaluateLogic(expression: Any?, data: Any?): Any? =
+        ReplaceExecutor.buildFromExpression(expression.asList)?.execute()
+}
+
+private data class ReplaceExecutor(
+    val replaceCandidate: String,
+    val oldString: String,
+    val newString: String,
+    val mode: ReplaceMode
+) {
+
+    fun execute() = mode.execute(this)
+
+    companion object {
+        const val REPLACE_CANDIDATE_INDEX = 0
+        const val OLD_STRING_INDEX = 1
+        const val NEW_STRING_INDEX = 2
+        const val COUNT_INDEX = 3
+    }
+}
+private fun ReplaceExecutor.Companion.buildFromExpression(expression: List<Any?>): ReplaceExecutor? {
```

Comment:
imo declaring static extension stuff complicates things. I recommend moving this function to the replace object
```
object Replace : StandardLogicOperation, StringUnwrapStrategy {
    override fun evaluateLogic(expression: Any?, data: Any?): Any? =
        buildFromExpression(expression.asList)?.execute()

    private fun buildFromExpression(expression: List<Any?>): ReplaceExecutor? =
        runCatching {
            val mode = ReplaceMode.build(expression[COUNT_INDEX] as String)
            ReplaceExecutor(
                expression[REPLACE_CANDIDATE_INDEX] as String,
                expression[OLD_STRING_INDEX] as String,
                expression[NEW_STRING_INDEX] as String,
                mode
            )

        }.fold(
            onSuccess = { it },
            onFailure = { null }
        )
}
```

Reply:
going this way also consts may be moved there to avoid unnecessary imports